### PR TITLE
Add support for gitpod

### PR DESCRIPTION
**Pull request by jrderuiter**
This PR proposes to add support for Gitpod, which can provide (new) contributors with one-click environment to use for working on the pandas code base. 

The PR includes two main additions:
- .gitpod.yml - Config file for gitpod that specifies which image/Dockerfile to use for gitpod and which init command(s) to run when booting the pod.
- .gitpod.Dockerfile - Dockerfile used to build a base image for the pod. Partially adapted from the existing Dockerfile in the repo.

Some things maybe still to consider:
- Gitpod has some options for providing pre-built workspaces (https://www.gitpod.io/docs/prebuilds/). This may be interesting to enable.
- There is some duplication between the two dockerfiles, but unfortunately the VS Code Dockerfile is not entirely compatible with the gitpod one. As such, the duplication may not be a bad thing. 

- [x] closes #xxxx (N/A)
- [x] tests added / passed (N/A)
- [x] passes `black pandas` (N/A)
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff` (N/A)
- [x] whatsnew entry (N/A)

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+image:
+  file: .gitpod.Dockerfile
+tasks:
+  - init: pip install -r requirements-dev.txt && make develop
```
Comment (WillAyd) [inline .gitpod.yml:4]: Can you use conda here instead of pip?

File: .gitpod.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+USER root
+
+RUN apt-get update \
+    && apt-get -y install --no-install-recommends apt-utils dialog 2>&1 \
+    #
+    # Verify git, process tools, lsb-release (common in install instructions for CLIs) installed
+    && apt-get -y install git iproute2 procps iproute2 lsb-release \
+    #
+    # Install C compilers (gcc not enough, so just went with build-essential which admittedly might be overkill),
```
Comment (WillAyd) [inline .gitpod.Dockerfile:11]: If you can get conda to work then can just install the compiler from conda-forge

https://anaconda.org/conda-forge/c-compiler

File: .gitpod.yml (hunk 1 above)
Comment (jrderuiter) [reply .gitpod.yml:4]: Sure, any reason to prefer conda over pip? Is that the preferred development environment for pandas?

File: .gitpod.Dockerfile (hunk 2 above)
Comment (jrderuiter) [reply .gitpod.Dockerfile:11]: This is copied from the existing Dockerfile, so I thought it would be ok?

**Comment by WillAyd** [comment]
Didn't realize this was copied from an existing Dockerfile. With that being the case, any reason to not just leverage the existing file? Looks like you can do that in a config

https://www.gitpod.io/docs/config-docker/

**Comment by MarcoGorelli** [comment]
Thanks @jrderuiter - I wasn't aware of gitpod, but this looks really cool!

However, when I clicked on "gitpod - ready to code" from your profile, I get an environment that doesn't look quite set up correctly:
```python-traceback
gitpod /workspace/pandas $ python -c "import pandas"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/workspace/pandas/pandas/__init__.py", line 16, in <module>
    raise ImportError(
ImportError: Unable to import required dependencies:
numpy: No module named 'numpy'
pytz: No module named 'pytz'
```

Is there any extra configuration required? And as Will suggested, does it work if you use the existing Dockerfile in
```yaml
image:
  file: .gitpod.Dockerfile
```
?

**Comment by MarcoGorelli** [comment]
closing as stale, but please ping if you'd like to keep working on this

**Pull request closed without merging**